sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,654 @@
+        Attribute to access any fitted sub-estimators by name.
+
+    final_estimator_ : estimator
+        The classifier which predicts given the output of `estimators_`.
+
+    stack_method_ : list of str
+        The method used by each base estimator.
+
+    References
+    ----------
+    .. [1] Wolpert, David H. "Stacked generalization." Neural networks 5.2
+       (1992): 241-259.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.svm import LinearSVC
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.ensemble import StackingClassifier
+    >>> X, y = load_iris(return_X_y=True)
+    >>> estimators = [
+    ...     ('lr', LogisticRegression(solver='lbfgs', multi_class='auto',
+    ...                               tol=1e-1)),
+    ...     ('svr', LinearSVC(tol=1e-1, random_state=42))
+    ... ]
+    >>> clf = StackingClassifier(
+    ...     estimators=estimators,
+    ...     final_estimator=RandomForestClassifier(n_estimators=10,
+    ...                                            random_state=42),
+    ...     cv=5,
+    ... )
+    >>> from sklearn.model_selection import train_test_split
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+    ... X, y, stratify=y, random_state=42
+    ... )
+    >>> clf.fit(X_train, y_train).score(X_test, y_test)
+    0.9...
+
+    """
+    def __init__(self, estimators, final_estimator=None, cv=None,
+                 stack_method='auto', n_jobs=None, verbose=0):
+        super().__init__(
+            estimators=estimators,
+            final_estimator=final_estimator,
+            cv=cv,
+            stack_method=stack_method,
+            n_jobs=n_jobs,
+            verbose=verbose
+        )
+
+    def _validate_meta_estimator(self):
+        super()._validate_final_estimator(default=LogisticRegression())
+        if not is_classifier(self.final_estimator_):
+            raise ValueError(
+                "'final_estimator' parameter should be a classifier. Got {}"
+                .format(self.final_estimator_)
+            )
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like of shape (n_samples,) or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+        check_classification_targets(y)
+        self._le = LabelEncoder().fit(y)
```

Comment:
Is this needed or are we just trying to remove redundancies here? All the models can to that themselves right?

Reply:
It is required to pass the common test and having the proper `self.classes_`